test/transformers/test_tiled_mlp.py

```diff
@@ -0,0 +1,302 @@
+
+from transformers.models.llama.configuration_llama import LlamaConfig
+
+from liger_kernel.transformers.geglu import LigerGEGLUMLP
+from liger_kernel.transformers.swiglu import LigerSwiGLUMLP
+from liger_kernel.transformers.tiled_mlp import LigerTiledGEGLUMLP
+from liger_kernel.transformers.tiled_mlp import LigerTiledSwiGLUMLP
+from liger_kernel.utils import infer_device
+
+device = infer_device()
+
+LLAMA_GEGLU_CONFIG = LlamaConfig(
+    hidden_size=1024,
+    intermediate_size=2048,
+    hidden_act="gelu_pytorch_tanh",
+)
+
+LLAMA_SWIGLU_CONFIG = LlamaConfig(
+    hidden_size=1024,
+    intermediate_size=2048,
+    hidden_act="silu",
+)
+
+
+@pytest.mark.parametrize(
+    "bsz, seq_len, hidden_size, intermediate_size",
+    [
+        (2, 512, 512, 1024),
+        (1, 1024, 256, 512),
+        # weird shapes
+        (4, 127, 128, 256),
+    ],
+)
+@pytest.mark.parametrize(
+    "dtype, atol, rtol",
+    [
+        # Tiled computation reorders operations, leading to numerical differences
+        # Larger tolerances account for accumulated floating-point errors
+        (torch.float32, 1.0, 1e-2),
+        # bfloat16 tests are skipped due to large numerical differences from tiling
+        # This is expected behavior as bfloat16 has lower precision
+        pytest.param(
+            torch.bfloat16,
+            100.0,
+            1.0,
+            marks=pytest.mark.skip(reason="bfloat16 has too much accumulated error with tiling"),
+        ),
+    ],
+)
+@pytest.mark.parametrize("num_shards", [None, 2, 4])
+def test_tiled_geglu_correctness(bsz, seq_len, hidden_size, intermediate_size, dtype, atol, rtol, num_shards):
+    """Test that TiledGEGLUMLP produces similar results as regular GEGLUMLP (float32 only)."""
+    config = LlamaConfig(
+        hidden_size=hidden_size,
+        intermediate_size=intermediate_size,
+        hidden_act="gelu_pytorch_tanh",
+    )
+
+    _input = torch.randn(bsz, seq_len, hidden_size, device=device, dtype=dtype)
+
+    x1 = _input.clone().requires_grad_(True)
+    x2 = _input.clone().requires_grad_(True)
+
+    # Initialize weights
+    G = torch.randn(hidden_size, intermediate_size, device=device, dtype=dtype)
+    U = torch.randn(hidden_size, intermediate_size, device=device, dtype=dtype)
+    D = torch.randn(intermediate_size, hidden_size, device=device, dtype=dtype)
+
+    # Regular GEGLU MLP
+    regular_mlp = LigerGEGLUMLP(config=config).to(device).to(dtype)
+    regular_mlp.gate_proj.weight.data = G.T
+    regular_mlp.up_proj.weight.data = U.T
+    regular_mlp.down_proj.weight.data = D.T
+
+    # Tiled GEGLU MLP
+    tiled_mlp = LigerTiledGEGLUMLP(config=config, num_shards=num_shards).to(device).to(dtype)
+    tiled_mlp.gate_proj.weight.data = G.T
+    tiled_mlp.up_proj.weight.data = U.T
+    tiled_mlp.down_proj.weight.data = D.T
+
+    # Forward pass
+    y1 = regular_mlp(x1)
+    y2 = tiled_mlp(x2)
+
+    assert torch.allclose(y1, y2, atol=atol, rtol=rtol), "Forward outputs don't match"
```

Comment:
nit: I personally prefer `torch.testing_assert_close()`. It has better error message, easier to debug.
```suggestion
   torch.testing.assert_close(y1, y2, atol=atol, rtol=rtol, msg=lambda x: "Forward outputs don't match\n" + x) 
```